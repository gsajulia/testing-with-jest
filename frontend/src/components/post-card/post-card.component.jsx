/* Styles */
import './post-card.styles.css';

export const PostCard = ({title, cover, body, id}) => (
        <div className="post">
            <img src={cover} alt={title} />
            <div className="post-content">
                <h1>
                    <b>{id} - Personagem:</b>
                     {title}
                </h1>
                <p>{body}</p>
            </div>
        </div>
)