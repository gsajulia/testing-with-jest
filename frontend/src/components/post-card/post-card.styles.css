
.post {
    background: #fff;
    box-shadow:  0 0 10px rgba(0, 0, 0, 0.1);
    transition: transform 100ms ease-in-out;
    display: flex;
    align-items: center;
    flex-direction: column;
  }
  
  .post:hover {
    transform: scale(1.05);
  }
  
  .post img {
    width: 100%;
    height: 450px;
  }
  
  .post-content {
    padding: 30px;
  }