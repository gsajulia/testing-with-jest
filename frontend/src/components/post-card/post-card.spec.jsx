import { render, screen } from '@testing-library/react';
import { PostCard } from './post-card.component';

const props = {
    title: 'Harry Potter',
    body: 'This character interpeted by Daniel Radcliffe is a human of Gryffindor house studant at Hogwarts and born at 31-07-1980 he is half-blood, has green eyes and black hair, his patronus is a stag and his wand core is phoenix feather and is made with holly wood.',
    id: 1,
    cover: 'img/img.png'
};

describe('<PostCard />', () => {
    it('should render PostCard correctly', () => {
      render(<PostCard {...props} />);
  
      expect(screen.getByAltText('Harry Potter'))
        .toHaveAttribute('src', 'img/img.png');
      expect(screen.getByRole('heading', { name: '1 - Harry Potter' })).toBeInTheDocument();
      expect(screen.getByText(/This character interpeted by Daniel Radcliffe is a human of Gryffindor house studant at Hogwarts and born at 31-07-1980 he is half-blood, has green eyes and black hair, his patronus is a stag and his wand core is phoenix feather and is made with holly wood./i))
        .toBeInTheDocument();
    });

    it('should match snapshot', () => {
      const { container } = render(<PostCard {...props} />);
      expect(container.firstChild).toMatchSnapshot();
    });
}); 
