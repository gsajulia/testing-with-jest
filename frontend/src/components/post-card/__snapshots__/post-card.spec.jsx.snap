// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Harry Potter"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h1>
      <b>
        1
         - 
      </b>
      Harry Potter
    </h1>
    <p>
      This character interpeted by Daniel Radcliffe is a human of Gryffindor house studant at Hogwarts and born at 31-07-1980 he is half-blood, has green eyes and black hair, his patronus is a stag and his wand core is phoenix feather and is made with holly wood.
    </p>
  </div>
</div>
`;
